File: final-api/src/main/java/com/salesianostriana/dam/finalapi/dtos/site/CreateSiteDto.java
package com.salesianostriana.dam.finalapi.dtos.site;

import lombok.*;
import org.hibernate.validator.constraints.URL;

import javax.persistence.Lob;
import javax.validation.constraints.*;
import java.time.LocalTime;

@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
@Builder
public class CreateSiteDto {

    @NotBlank(message = "{site.name.blank}")
    @Size(min = 2, max = 32, message = "{site.name.size}")
    private String name;
    @Lob
    private String description;
    @NotBlank(message = "{site.address.blank}")
    private String address;
    @NotBlank(message = "{site.city.blank}")
    private String city;
    @NotBlank(message = "{site.postalCode.blank}")
    @Size(min = 5, max = 5, message = "{site.postalCode.size}")
    private String postalCode;
    @NotBlank(message = "{site.email.blank}")
    @Email(message = "{site.email.email}")
    private String email;
    @NotBlank(message = "{site.phone.blank}")
    @Size(min = 9, max = 9, message = "{site.phone.size}")
    private String phone;
    @URL(message = "{site.url.url}")
    private String web;
    @NotBlank(message = "{site.openingHour.blank}")
    @Min(value = 0, message = "{site.openingHour.min}")
    @Max(value = 23, message = "{site.openingHour.max}")
    private LocalTime openingHour;
    @NotBlank(message = "{site.closingHour.blank}")
    @Min(value = 0, message = "{site.closingHour.min}")
    @Max(value = 23, message = "{site.closingHour.max}")
    private LocalTime closingHour;
}

File: flutter_app/lib/models/create_site_dto.dart
class CreateSiteDto {
  CreateSiteDto({
    required this.name,
    required this.description,
    required this.address,
    required this.city,
    required this.postalCode,
    required this.email,
    required this.phone,
    required this.web,
    required this.openingHour,
    required this.closingHour,
  });
  late final String name;
  late final String description;
  late final String address;
  late final String city;
  late final String postalCode;
  late final String email;
  late final String phone;
  late final String web;
  late final DateTime openingHour;
  late final DateTime closingHour;

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['name'] = name;
    data['description'] = description;
    data['address'] = address;
    data['city'] = city;
    data['postalCode'] = postalCode;
    data['email'] = email;
    data['phone'] = phone;
    data['web'] = web;
    data['openingHour'] = openingHour;
    data['closingHour'] = closingHour;
    return data;
  }
}
